apps/web/app/notFoundClient.tsx

```diff
@@ -57,6 +58,27 @@ export function NotFound() {
   const isSubpage = pathname?.includes("/", 2) || isBookingSuccessPage;
   const isInsights = pathname?.startsWith("/insights");
 
+  const [redirected, setRedirected] = useState(false);
+
+  useEffect(() => {
```

Comment:
Consider using React Query for this data fetching operation instead of raw fetch in useEffect. This would align with the rest of the application's data fetching patterns and provide better handling for loading states, caching, and error management. Something like:

```typescript
const { data } = useQuery({
  queryKey: ['previousUsername', username],
  queryFn: () => fetch(`/api/users/username/${username}?checkPrevious=true`).then(res => res.json()),
  enabled: pageType === PageType.USER && !isSubpage && !!username,
  onSuccess: (data) => {
    if (data.currentUsername) {
      window.location.href = `/${data.currentUsername}`;
    }
  }
});
```